### ISSUE TITLE
Model enums are broken in generated C# client code

### ISSUE THREAD

**Issue by JColl88** [swagger-api/swagger-codegen#8027]
##### Description

(This issue my be related to #7947.)

I have tried to generate a C# client solution using the codegen-cli 2.3.1.

The specification in question can be found at: https://api.apis.guru/v2/specs/amazonaws.com/AWSMigrationHub/2017-05-31/swagger.json

This functions fine with 2.2.3, and generates a solution that can be built in VS2013. However, using 2.3.1, the solution cannot be built as the C# code in one of the model classes (Status.cs) is invalid. A snippet of the class is shown below which illustrates the problem:

Status.cs:34 (generated using 2.2.3) - this works as expected:
```        /// <summary>
        /// Enum NOTSTARTED for "NOT_STARTED"
        /// </summary>
        [EnumMember(Value = "NOT_STARTED")]
        NOTSTARTED,
```

Status.cs:34 (generated using 2.3.1) - this generates invalid C#:
```        /// <summary>
        /// Enum NOTSTARTED for value: NOT_STARTED
        /// </summary>
        
        NOTSTARTED = NOT_STARTED,
```

If I generate using 2.3.1, but specify a folder containing the modelEnum.mustache template from 2.2.3, the following is generated, which is still invalid:
```        /// <summary>
        /// Enum NOTSTARTED for NOT_STARTED
        /// </summary>
        [EnumMember(Value = NOT_STARTED)]
        NOTSTARTED,
```

(where the quotes around NOT_STARTED have not been planted.)

As of 16/04/18, the issue is still unfixed in the 2.4 snapshot.

##### Swagger-codegen version

2.3.1

##### Swagger declaration file content or url

https://api.apis.guru/v2/specs/amazonaws.com/AWSMigrationHub/2017-05-31/swagger.json

##### Command line used for generation

```java -jar swagger-codegen-cli-2.3.1.jar generate -i aws_swagger.json -l csharp -DapiTests=false -DmodelTests=false```

(where aws_swagger.json is a local copy of the version linked above.)

##### Steps to reproduce

Run the above command for the specification linked above (AWS Migration Hub).

##### Related issues/PRs

#7947

##### Suggest a fix/enhancement

In the spec, the Status enum is defined as follows:
```    "Status": {
      "enum": [
        "NOT_STARTED",
        "IN_PROGRESS",
        "FAILED",
        "COMPLETED"
      ],
      "type": "string"
```

By placing dummy text in isString tags in the modelEnum.mustache template (e.g. {{#isString}}This is a string{{/isString}}), it becomes apparent that the type tag for this variable is not being set correctly (to string), as the dummy text is not planted in the generated code.

**Comment by jages** [comment]
Building 'swagger-codegen' locally, i have found a makeshift solution for this problem (that obviously requires further testing).
[src/main/resources/csharp/enumClass.mustache](https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/csharp/enumClass.mustache) @ line 14
replace 
`[EnumMember(Value = {{#isLong}}"{{/isLong}}{{#isInteger}}"{{/isInteger}}{{#isFloat}}"{{/isFloat}}{{#isDouble}}"{{/isDouble}}{{{value}}}{{#isLong}}"{{/isLong}}{{#isInteger}}"{{/isInteger}}{{#isDouble}}"{{/isDouble}}{{#isFloat}}"{{/isFloat}})]`

with 
`[EnumMember(Value = "{{{value}}}")]`

This works for the [petstore/v2-demo](http://petstore.swagger.io/v2/swagger.json), but needs to be looked into by someone with more experience. i don't think there is any DataType that cannot be masked with double quotes.

**Comment by coin8086** [comment]
I also have this problem. Please fix is ASAP. 

BTW, I found that the enum types are correct when download a generated client via the SwaggerHub(https://app.swaggerhub.com) online, but using 'swagger-codegen' 2.3.1 locally produces incorrect code. So the problem must be fixed officially already. 

Please help.

**Comment by codersam** [comment]
Same problem in version 3.0.14, installed from Homebrew.

**Comment by frantuma** [comment]
should be fixed in #9936 (and swagger-api/swagger-codegen-generators#561 for 3.0.0)
